src/main/java/seedu/address/model/travelplanner/ModelManager.java

```diff
@@ -151,62 +150,68 @@ public void setActivity(Activity target, Activity editedActivity) {
     @Override
     public void setDirectory(Directory dir) {
         directory = dir;
-        setCurrentTravelPlan();
     }
 
     @Override
     public Directory getDirectory() {
         return directory;
     }
 
-    /**
-     * Sets the current travel plan to the current directory.
-     */
-    private void setCurrentTravelPlan() {
-        if (directory.isTravelPlan()) {
-            currentTravelPlan = (TravelPlan) directory;
-            currentFilteredTravelPlanObjectList = new FilteredList<>(currentTravelPlan.getActivityList());
-        }
-    }
+    //========== TravelPlanObjectList ===================================================
 
     @Override
     public void setToActivityList() {
-        currentFilteredTravelPlanObjectList = new FilteredList<>(currentTravelPlan.getActivityList());
+        currentFilteredTravelPlanObjectList = new FilteredList<>(directory.getActivityList());
     }
 
     @Override
     public void setToAccommodationList() {
-        currentFilteredTravelPlanObjectList = new FilteredList<>(currentTravelPlan.getAccommodationList());
+        assert directory instanceof TravelPlan
+                : "Directory must be set to a TravelPlan to call setToAccommodationList.";
+        TravelPlan tp = (TravelPlan) directory;
```

Comment:
Is there a better way of doing this? Like not typecasting but instead use the polymorphism.

Reply:
I think this is still considered polymorphism though?

Reply:
thats why before i put 2 boolean method to check is which one

Reply:
if want to check can use instanceof!

Reply:
nah typecasting is not considered polymorphism. We have to do directory.the method straight away i think. But it might be not worth the time to think about that. So maybe can leave it like this first ba

Reply:
ok!